### PR TITLE
[DCP] Add a profiler function for benchmarking save and load

### PR DESCRIPTION
**Pull request by fegin**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116248
* #116247
* #116246
* #116245
* __->__ #116007
* #116006

Many operations when calling DCP's save and load are executed on CPU. Thus we can easily profile these operations with cProfile. This PR adds the ability to profile the save() and load()

One follow-up for this PR is to integrate the feature with the distributed logging flags.

Differential Revision: [D52245434](https://our.internmc.facebook.com/intern/diff/D52245434/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @LucasLLC

### CODE REVIEW

File: torch/distributed/checkpoint/utils.py [hunk 1]
```diff
@@ -371,3 +374,22 @@ def _normalize_device_info(device_type: str, device_id: int) -> str:
     if device_type == "cpu":
         return "cpu"
     return f"{device_type}:{device_id}"
+
+
+# TODO: integrate with distributed logging flag
+ENABLE_PROFILE = False
```
Comment (wz337) [inline torch/distributed/checkpoint/utils.py:380]: Just to make sure I understand correctly. The default is without the profiling unless manually change `ENABLE_PROFILE` to True. Is that correct?

**Review by LucasLLC — APPROVED** [review]
Super helpful, thanks!

File: torch/distributed/checkpoint/utils.py (hunk 1 above)
Comment (fegin) [reply torch/distributed/checkpoint/utils.py:380]: Yes, we don't want to enable this by default. We can integrate it with torch distributed logging later.

**Review by wz337 — APPROVED** [review]
LGTM!

**Pull request closed without merging**